eth/handler.go

```diff
@@ -444,11 +444,9 @@ func (pm *ProtocolManager) handleMsg(p *peer) error {
 			// Irrelevant of the fork checks, send the header to the fetcher just in case
 			headers = pm.fetcher.FilterHeaders(headers, time.Now())
 		}
-		if len(headers) > 0 || !filter {
```

Comment:
No, this is not always true, because `headers` is changed 2 lines above, and `len(headers)` becomes `0`, while `filter` was `true` (hence `!filter` false).